src/OrchardCore/OrchardCore.Mvc.Core/Extensions/OrchardCoreBuilderExtensions.cs

```diff
@@ -15,6 +15,13 @@ public static OrchardCoreBuilder AddMvc(this OrchardCoreBuilder builder)
             {
                 // Allows a tenant to add its own route endpoint schemes for link generation.
                 collection.AddSingleton<IEndpointAddressScheme<RouteValuesAddress>, ShellRouteValuesAddressScheme>();
+
+                collection.Configure<RouteOptions>(options =>
+                {
+                    // The Cors module is designed to handle CORS, thus we skip checking for unhandled security metadata by default.
+                    // Additionally, skipping security metadata checks on the endpoint provides a minor performance benefit.
+                    options.SuppressCheckForUnhandledSecurityMetadata = true;
```

Comment:
I might have another fix since this removes a check for a misconfigured application.

Reply:
How could OC end up being misconfigured? 

We could configure this using a configuration provider and then set it to `false` just to ensure that all tests pass with it being false but use true by default

Reply:
Seems I forgot to reply to this, it's something related to what @gvkries refers to in https://github.com/OrchardCMS/OrchardCore/pull/16437#issuecomment-2228240651